### PR TITLE
Implement REST API

### PR DESCRIPTION
**Pull request by kuzmavictor**
During this iteration was implemented:
- the service that provides all REST methods (the class that implements it is located on the following path: `name_application\servise\request-service` );
- all method for `FileListService` class (the class that implements it is located on the following path: `name_application\servise\files` ); 
- the necessary functionality inside  (`ServiceProvider`, `AuthorizaionService`, `RegistrationService`, `FileListService`);
- the models of file and user-info.

### CODE REVIEW

**Comment by kuzmavictor** [comment]
@dkalinin PTAL.

File: app/application/index.js [hunk 1]
```diff
@@ -31,7 +33,7 @@ export class Application {
    * @private
    */
   _configRouter() {
-    const requestServiceProvider = new RequestServiceProvider();
+    const requestServiceProvider = new RequestServiceProvider(ROOT_APPLICATION_URL);
     const store = new Store(requestServiceProvider);
```
Comment (dkalinin) [inline app/application/index.js]: `store` and `requestServiceProvider` should be created in other place than `_configRouter` method.

File: app/controller/authorization/index.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ export class AuthorizationController extends BaseController {
       return;
     }
     this.authorizationInProgress = true;
-    const promise = this._authorizationService.doPost('/toLogin', authorizationDTO);
+    const promise = this._authorizationService.logIn('/toLogin', authorizationDTO);
```
Comment (dkalinin) [inline app/controller/authorization/index.js]: /toLogin -> URL is a responsibility of _authorizationService.

File: app/model/file-model/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/**
+ * This class  represents a file model.
+ */
+export class FileModel {
+  /**
+   * The constructor sets fields values  of {@link FileModel}.
+   *
+   * @param {Object} file -an object that contains data abaut a file in hash, that object
```
Comment (dkalinin) [inline app/model/file-model/index.js]: abaut -> about

File: app/service/files/index.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    });
+  }
+
+  /**
+   * Emulates process of getting user info.
+   *
+   * @param {String} userId - an user id
+   * @return {Promise<any>}
+   */
+  getUserInfo(userId) {
```
Comment (dkalinin) [inline app/service/files/index.js]: Let's move to user service.

File: app/service/files/index.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+  // TODO:2019-09-07:Victor.Kuzma: After the implementation
+  //  of server-side this method will be deleted.
+  /**
+   * It is a temporary method. Returns object that contains stub folder (emulates data from server).
+   * {@link FileListService.getFiles}. (It's a stub method).
+   *
+   * @return {FileModel} - a file model.
+   * @private
+   */
+  static _getStubFilder() {
```
Comment (dkalinin) [inline app/service/files/index.js]: _getStubFilder -> _getStubFolder

**Pull request merged**